### PR TITLE
Add more test cases in BigQueryToTable

### PR DESCRIPTION
**Pull request by boyuanzz**

### CODE REVIEW

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
R: @pabloem 
This PR added more test cases for BigQueryToTableIT

**Comment by boyuanzz** [comment]
NOTE: All gradle build file changes will be reverted before merge.

**Comment by boyuanzz** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
@chamikaramj 
Hey Cham, do you think it's worthy for us to add these more bigquery IT test cases?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java [hunk 1]
```diff
@@ -60,9 +65,135 @@
   private BigQueryOptions bqOption;
   private String outputTable;
   private BigqueryClient bqClient;
+  private final TableSchema legacyQueryTableSchema =
```
[the remark this answers is not in the document]
Comment (boyuanzz) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: hey, seems like, we only use uppercase naming when it'a a static final string, for example: [code sample](https://github.com/apache/beam/blob/279a05604b83a54e8e5a79e13d8761f94841f326/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PropertyNames.java#L22). For static string, seems like following normal naming rules: [example](https://github.com/apache/beam/blob/279a05604b83a54e8e5a79e13d8761f94841f326/runners/samza/src/main/java/org/apache/beam/runners/samza/adapter/UnboundedSourceSystem.java#L191). wdyt?

[the remark this answers is not in the document]
Comment (boyuanzz) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: For this case to my knowledge, it's probably okay to use these constant string in a non-static way right?

[the remark this answers is not in the document]
Comment (pabloem) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: Yes, but `static final` is the convention for global constants. The second example that you show is a per-instance variable that comes in through the constructor.

In this case, I believe `legacyQueryTableSchema` is a constant for every test right?

See [here](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java#L177) for a non-string constant with uppercase naming.

[the remark this answers is not in the document]
Comment (pabloem) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryToTableIT.java]: If `legacyQueryTableSchema` is not a constant for the whole test file, then yes, you can leave it as-is : )

**Comment by chamikaramj** [comment]
Is there any downside to adding more tests (Are the flaky for example) ? If not +1 for adding more tests.

**Comment by chamikaramj** [comment]
Also looks like tests cover different aspects (legacy vs standard SQL for example).

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
I checked the test results. Seems like each test was ran twice, and two failure tests had 1 passed and 1 failed: https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild_PR/117/testReport/org.apache.beam.sdk.io.gcp.bigquery/BigQueryToTableIT/

**Comment by boyuanzz** [comment]
The IT tests passed when running locally.

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by pabloem** [comment]
One of the tests seems to have failed. I'll rerun, and if not I believe it must be fine to `@Ignore` it and add a todo assigned to me to fix.

**Comment by pabloem** [comment]
Run Java PostCommit

**Comment by boyuanzz** [comment]
Re: @pabloem 
seems like direct runner tests failed. I can exclude this test from direct runner for now and create a follow up JIRA. sounds good?

**Comment by boyuanzz** [comment]
Run Java PostCommit

**Comment by pabloem** [comment]
Yes, thank you Boyuan!

**Comment by udim** [comment]
Was there ever a JIRA created to make these tests work on DirectRunner?
Also, aren't DirectRunner-based I/O integration preferred over Dataflow since they run faster and we're only testing SDK code anyway?

**Comment by boyuanzz** [comment]
> Was there ever a JIRA created to make these tests work on DirectRunner?
There is no JIRA for DirectRunner.
> Also, aren't DirectRunner-based I/O integration preferred over Dataflow since they run faster and we're only testing SDK code anyway?
> When I creating these test cases, I only tested on DataflowRunner. I'm not sure whether they work on DirectRunner well, but it's very easy to add this test suite to DirecrRunner test: https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/build.gradle#L101

**Pull request merged**